### PR TITLE
gmp: use --build configure option

### LINKED ISSUE #7054: GMP was compiled with -march=sandybridge
I use a library that decided to reuse the compilation flags of gmp that are defined in __GMP_CFLAGS at the end of gmp.h. If gmp was installed as binary package via homebrew, these flags contain -march=sandybridge and -mtune=sandybridge.

Given that there are still machines out there with pre-sandybridge processors, this seems very risky. Could we change this to some sane defaults? Debian, for example, does not specify -mtune and -march at all.

For some reason, I have never encountered a problem with gmp itself. Is gmp even compiled with these options? Or are they just spurious remainders of the machine that builds the packages?

### PR DESCRIPTION
**Pull request by ilovezfs**
and remove --disable-assembly.

CC @smtrat @MikeMcQuaid @fxcoudert

Closes #7054.

### CODE REVIEW

File: Formula/gmp.rb [hunk 1]
```diff
@@ -16,12 +16,13 @@ class Gmp < Formula
 
   def install
     ENV.cxx11 if build.cxx11?
-    args = ["--prefix=#{prefix}", "--enable-cxx"]
 
-    # https://github.com/Homebrew/homebrew/issues/20693
-    args << "--disable-assembly" if build.bottle?
+    # XNU release major version
+    osmajor = MacOS.version.to_s[3..-1].to_i + 4
```
Comment (MikeMcQuaid) [inline Formula/gmp.rb]: What does the output end up being? Should rather shell out for this if we can; fairly opaque as-is (not your fault)

File: Formula/gmp.rb [hunk 2]
```diff
@@ -16,12 +16,13 @@ class Gmp < Formula
...
 
-    system "./configure", *args
+    system "./configure", "--prefix=#{prefix}",
+                          "--enable-cxx",
+                          "--build=core2-apple-darwin#{osmajor}"
```
Comment (MikeMcQuaid) [inline Formula/gmp.rb]: Let's just do this when `build.bottle?`

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/gmp.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/gmp.rb]: This is how XNU is versioned
10.6 -> 10
10.7 -> 11
10.8 -> 12
10.9 -> 13
...
10.12 -> 16

There's no reason to shell out if we know the answer.

Comment (MikeMcQuaid) [inline Formula/gmp.rb]: Other formula generally use for this
```
ruby
`uname -r`.chomp
```
I'd be 👍 on adding a Homebrew/brew method that exposed this too.

Comment (ilovezfs) [inline Formula/gmp.rb]: That's the full version. `uname -r`.chomp
=> "12.6.0"

So you still end up having to do string manipulations then to extract the major version. Running processes for answers we already know is pretty gross lol

**Comment by ilovezfs** [comment]
pr refreshed with useless shell invocation

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request closed without merging**